python/lsst/pipe/base/_dataset_handle.py

```diff
@@ -49,7 +50,7 @@ def __init__(
         *,
         storageClass: StorageClass | None = None,
         parameters: dict[str, Any] | None = None,
-        dataId: dict[str, Any] | DataCoordinate | None = None,
+        dataId: dict[str, Any] | DataCoordinate | Mapping | None = None,
```

Comment:
I guess this is not to allow arbitrary `Mapping[Any, Any]` but just to remove `type: ignore`? Could it be made `Mapping[str, Any]`? You could probably drop `dict` then.

Reply:
I added `Mapping` here because `frozendict` is a `Mapping`. I hadn't spotted that I already had `dict` in there so I can replace that with Mapping.

Reply:
I changed it to `Mapping[Any, Any]` and dropped the separate `dict`.

Reply:
I'm so slow catching up. Sorry. I see that that is exactly what `DataId` is defined as (without the `None`).